### DISCUSSION TITLE
Local compile instead of gitpod

### DISCUSSION THREAD

**Discussion by teaalltr** [benzino77/tasmocompiler#389] [Q&A]
Is it possible?

**Comment by benzino77** [comment]
Sure. There is a description in [README](https://github.com/benzino77/tasmocompiler#easy-way) file.
You can run it locally using docker or compile it from the source (both methods are described in the linked documentation)

**Reply by akula3** [reply]
Hi,
I'm also looking for local compilation with PlatformIO.
Your pointed information in the README made me not wiser :-(
May you please elaborate a bit deeper?

**Reply by benzino77** [reply]
What operating system you are running on?

**Reply by akula3** [reply]
I'm on Win11

**Reply by benzino77** [reply]
So what you need to do is install [Docker Desktop](https://www.docker.com/products/docker-desktop/) or [Rancher Desktop ](https://rancherdesktop.io/) and then run, provided in README, docker command in your terminal

**Reply by akula3** [reply]
Thanks for the reply!
I'm using PlatformIO and I would stay on it. Therefore I prefer [Less easy way](https://github.com/benzino77/tasmocompiler?tab=readme-ov-file#less-easy-way) which, I understand, runs under PlatformIO. Would you be so kind to give some advice on that?

**Reply by benzino77** [reply]
TasmoCompiler is using PlatformIO under the hood.
If you want to use "Less easy way" method, you need to ensure all needed programs and libraries (nodejs, python, platformio) are installed on your system.

To install python follow the instructions provided [here](https://www.python.org/downloads/)
To install NodeJS follow the instructions provided [here](https://nodejs.org/en/download/prebuilt-binaries) 
To install platformio you need to type `pip3 install platformio`
Download TasmoCompiler source code and extract it
![Screenshot from 2024-06-22 08-28-47](https://github.com/benzino77/tasmocompiler/assets/15658235/03311fd8-f24a-4b60-8db4-ba7e95752c40)

Open your terminal and change your working directory to the directory where TasmocCompiler was exctracted. Then type in your terminal:
```sh
npm install
npm build
node server/app.js
```
point your browser to http://localhost:3000/


But really, using docker nowadays is the fastest, most convenient, beginners friendly way. I encourage you to seriously consider this method ;)

**Reply by Noschvie** [reply]
`But really, using docker nowadays is the fastest, most convenient, beginners friendly way.`

@akula3 Have a look to docker and compose and you will have fun, it's really easy tu use!

**Reply by akula3** [reply]
Hey @benzino77 and @Noschvie ,
thanks for information and motivation :-)
Maybe I'll give a try to docker.
